### ISSUE TITLE
Support Nonlinear Expressions

### ISSUE THREAD

**Issue by Downsite** [jump-dev/Gurobi.jl#556]
Gurobi 11 now support directly addressing nonconvex nonlinear constraints with spacial branch and bound.
It would be great to test Gurobi with the new first-class nonlinear modeling in MathOptInterface.

Older versions already supported "Function Constraints" with  piece-wise linear approximations, but that would have been hard to bring to users transparently.

From the discussion [here](https://discourse.julialang.org/t/cant-solve-a-problem-with-nonlinear-terms-like-x-0-7-or-x-3-using-jump-with-gurobi-11/106896/8), it seems that a while ago, this was blocked by the API on the side of Gurobi, and we were waiting for "first-class nonlinear API".

Is the API listed [here](https://www.gurobi.com/documentation/current/refman/c_agc_xxx.html#routine:GRBaddgenconstrXxx) seems to be the one that the comment was based on and is already referenced in the documentation.
Is the API still blocking the adoption of nonlinear modeling?

**Comment by odow** [comment]
This is a duplicate of https://github.com/jump-dev/Gurobi.jl/issues/554.

You can add general constraints as documented in the README:
https://github.com/jump-dev/Gurobi.jl?tab=readme-ov-file#the-c-api-from-jump

**Comment by odow** [comment]
Closing as won't fix. We will address this once Gurobi supports a nonlinear expression interface (like was recently done in Xpress.jl: https://github.com/jump-dev/Xpress.jl/pull/255)
